### PR TITLE
Paginate records

### PR DESCRIPTION
**Pull request by jaskaransarkaria**
* paginate cases when there are more than the max page items (add pagination component)
* add e2e test
* add integration test
* Add pagination story book
* Use query params for pagination

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
```
Comment (jaskaransarkaria) [inline .storybook/main.js:1]: this is needed to handle component path aliases for storybook

File: setupTests.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (jaskaransarkaria) [inline setupTests.ts:1]: eslint complains that this should in dev dependencies

File: cypress/support/e2e.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@
 
 // Import commands.js using ES2015 syntax:
 import "./commands"
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (jaskaransarkaria) [inline cypress/support/e2e.ts:18]: eslint complains that this should in dev dependencies

File: .storybook/main.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+
 module.exports = {
   stories: ["../src/**/*.stories.mdx", "../src/**/*.stories.@(js|jsx|ts|tsx)"],
   addons: [
     "@storybook/addon-links",
     "@storybook/addon-essentials",
     "@storybook/addon-interactions",
-    "storybook-addon-designs"
+    "storybook-addon-designs",
+    "storybook-addon-next-router"
```
Comment (tomvaughan77) [inline .storybook/main.js:10]: Nice one 💯

**Review by tomvaughan77 — APPROVED** [review]

File: src/pages/index.tsx [hunk 5]
```diff
@@ -14,79 +14,91 @@ import Head from "next/head"
...
+      pageNum: validatedPageNum,
+      orderBy: validatedOrderBy,
+      order: validatedOrder
     })
 
     if (isError(courtCases)) {
       throw courtCases
     }
 
-    const oppositeOrder: QueryOrder = order === "asc" ? "desc" : "asc"
```
Comment (C-gyorfi) [inline src/pages/index.tsx:56]: Moving this before line 45, where we call `listCourtCases` causes a small bug, while the URL param shows asc/desc the return result is the opposite:
image

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**